chainermn/communicators/_memory_utility.py

```diff
@@ -70,8 +70,8 @@ def ptr(self):
     def buffer(self, size):
         return self.ffi.buffer(self.ffi.cast('void *', self.memory.ptr), size)
 
-    def array(self, shape, offset=0):
-        return cp.ndarray(shape, memptr=self.memory + offset, dtype=cp.float32)
+    def array(self, shape, offset=0, dtype=np.float32):
```

Comment:
So if we don't want cupy's default behaviour setting dtype as `float64` when `None` is passed, we might better add non-None assertion here.

Reply:
I I also think so, I will add the assertion.